### ISSUE TITLE
Support for small Twitter Card/FB images

### ISSUE THREAD

**Issue by adamgreenough** [sybrew/the-seo-framework#525]
Sometimes we don't want to blast feeds with huge images and would prefer to use the small square thumbnail. Uploading smaller photos, Twitter still picks up and stretches the small image to a large card.

Would be great to be able to choose or have it intelligently detect which card tags to put out based on image size!

**Comment by sybrew** [comment]
Yes, this has been bugging me for quite some time too. A per-post option would be great, where a global "auto-determine" setting would be neat.

For the intelligent part, see #428. With the new image parser brought in TSF v4.0, it should be much easier to implement.

Proposed global settings:
- Auto (explain...)
- Summary (summary)
- Large Image (summary_large_image)

Proposed per post settings:
- Default (based on global option, update value via AJAX on post-save?)
- Summary (summary)
- Large Image (summary_large_image)

**Comment by adamgreenough** [comment]
Seems like just a toggle in the social editor at a post level would be a great way to get this in fairly easily. Is this in the roadmap at all or should I look for alternative way of sorting it for time being? I think I could probably build a custom field toggle to filter the plugins output.

**Comment by sybrew** [comment]
There's no roadmap yet. However, I might squeeze it in the next major release, but that'll be a few months from now.

Until then, you can set the twitter card type with this unaptly named filter: `the_seo_framework_twittercard_output`.

This (untested) example filter sets the `$type` as `summary` when the image height is below 500px.

```php
add_filter( 'the_seo_framework_twittercard_output', function( $type, $id ) { 
	foreach ( the_seo_framework()->get_image_details_from_cache() as $image ) {
		if ( $image['height'] && $image['width'] ) {
			if ( $image['height'] < 500 ) {
				$type = 'summary';
			}
		}
		// Twitter Cards only use the first image provided.
		break;
	}

	return $type;
}, 10, 2 );
```

I'd love to see which logic you apply to determine the type output 😄

**Comment by sybrew** [comment]
Punting this issue and starting the release of 4.1.0--developer fatigue kicked in.

**Comment by sybrew** [comment]
This feature also affects how embeds are displayed on Discord. We may want to explain that in our settings.

**Comment by adamgreenough** [comment]
Still really wanting this feature! Surprised it's not had more interest, big cards can be obnoxious :)

**Comment by sybrew** [comment]
I'm forgoing the "auto" feature for now because:
1. What are the factors for making it "summary" versus "summary_large_image"?
2. What if there are video objects? Do we select Player (#76)?
3. What will be the fallback if no obvious filesize or dimensions are determinable? Do we add two types of "auto" options for the user?
4. I plan to have this update out in 1 week from now.

I will add a "default (%s)," "summary," and "summary_large_card" selector on a per-post basis for this update, so users have at least more control over this.

**Comment by adamgreenough** [comment]
Hoorah! Looking forward to it, I can delete another plugin when thats in, thank you!
